tests/test_spectrum_assignment.py

```diff
@@ -0,0 +1,378 @@
+        range2 = range(oms2.spectrum_bitmap.geti(nval2) - mval,
+                       oms2.spectrum_bitmap.geti(nval2) + mval -1)
+        for elem in range1:
+            print(f'value should be zero at index {elem}')
+            assert test2[elem] == 0
+
+        for elem in range2:
+            print(f'value should be zero at index {elem}')
+            assert test2[elem] == 0
+
+def test_values(setup):
+    """ checks that oms.assign_spectrum(13,7) is (193137500000000.0, 193225000000000.0)
+        reference to Recommendation G.694.1 (02/12), Figure I.3
+        https://www.itu.int/rec/T-REC-G.694.1-201202-I/en
+    """
+    network, oms_list = setup
+
+    oms_list[5].assign_spectrum(13, 7)
+    fstart, fstop = m_to_freq(13, 7)
+    print('expected: 193137500000000.0, 193225000000000.0')
+    print(f'obtained: {fstart}, {fstop}')
+    assert fstart == 193.1375e12 and fstop == 193.225 * 1e12
+    nstart = frequency_to_n(fstart)
+    nstop = frequency_to_n(fstop)
+    # nval, mval = slots_to_m(7, 20)
+    nval, mval = slots_to_m(nstart, nstop)
+    print('expected n, m: 13, 7')
+    print(f'obtained: {nval}, {mval}')
+    assert nval == 13 or mval == 7
+
+@pytest.mark.parametrize('nval', [0, None, 0.5])
+@pytest.mark.parametrize('mval', [1, 0, None, 4.5])
+def test_exception(nval, mval, setup):
+    """ test n or m not applicable values
+    """
+    network, oms_list = setup
+    try:
+        oms_list[5].assign_spectrum(nval, mval)
+        print(f'n, m values should raise an error {nval}, {mval}')
+        test = False
+    except SpectrumError:
+        test = True
+    print(nval, mval)
+    assert test or (nval + mval) == 1
+
+@pytest.mark.parametrize('nval', [0, -300, 500])
+@pytest.mark.parametrize('mval', [1, 600])
+def test_wrong_values(nval, mval, setup):
+    """ test n or m not applicable values
+    """
+    network, oms_list = setup
+    test = oms_list[5].assign_spectrum(nval, mval)
+    print(f'n, m values should raise an error {nval}, {mval}')
+    expected = False
+    if (nval + mval) == 1:
+        expected = True
+    print(nval, mval)
+    assert test is expected
+
+def test_bitmap_assignment(setup):
+    """ test that a bitmap can be assigned
+    """
+    network, oms_list = setup
+
+    oms_list[5].assign_spectrum(13, 7)
+
+    btmp = deepcopy(oms_list[5].spectrum_bitmap.bitmap)
+    freq_min = 191300000000000.0
+    freq_max = 196100000000000.0
+    # try a first assignment that must pass
+    spectrum_btmp = Bitmap(freq_min, freq_max, grid=0.00625e12, guardband=0.15e12, bitmap=btmp)
+
+    # try a wrong asignment that should not pass
+    btmp = btmp[1:-1]
+    test = False
+    try:
+        spectrum_btmp = Bitmap(freq_min, freq_max, grid=0.00625e12, guardband=0.15e12, bitmap=btmp)
+    except SpectrumError:
+        test = True
+
+    print('bitmap direct assignment should create an error if length is not consistant with' +\
+          'provided values')
```

Comment:
I think that it's better to use a longer line rather than split an error message into two just for two extra words. Let's address this once we move to Gerrit (with a testsuite's built-in PEP8 checking).

Reply:
OK!